assimilation_code/modules/utilities/mpi_utilities_mod.f90

```diff
@@ -127,9 +127,9 @@ module mpi_utilities_mod
 
 integer :: myrank        = -1  ! my mpi number
 integer :: total_tasks   = -1  ! total mpi tasks/procs
-integer :: my_local_comm =  0  ! duplicate communicator private to this file
-integer :: datasize      =  0  ! which MPI type corresponds to our r8 definition
-integer :: longinttype   =  0  ! create an MPI type corresponding to our i8 definition
+type(MPI_Comm) :: my_local_comm  ! duplicate communicator private to this file
```

Comment:
is there a way to give these an initial value that is not legal?  it would help to catch if they ever got used before being initialized.